### PR TITLE
feat: AWS Migration & OTA Capabilities added

### PR DESCRIPTION
**Pull request by yashawasthi10**

### CODE REVIEW

**Review by mohithemnani11 — APPROVED** [review]

File: api/test/redis-manager.ts [hunk 1]
```diff
@@ -2,128 +2,101 @@
...
+    assert(!this.body, "Body already set");
     this.body = body;
     return this;
   }
 
   public reset(): void {
-    delete this.statusCode;
-    delete this.body;
+    this.statusCode = undefined;
+    this.body = undefined;
```
Comment (arpitD11) [inline api/test/redis-manager.ts:27]: do we want to set this as undefined? why?

**Review by arpitD11 — CHANGES_REQUESTED** [review]

File: api/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  redis:
+    image: redis:alpine
+    ports:
+      - "6379:6379"  # Redis running on localhost:6379
+
+  localstack:
+    image: localstack/localstack
+    environment:
+      - SERVICES=s3
+      - AWS_DEFAULT_REGION=us-east-1
```
[the remark this answers is not in the document]
Comment (yashawasthi10) [reply api/docker-compose.yml]: Removed

File: api/script/default-server.ts [hunk 3]
```diff
@@ -2,19 +2,31 @@
...
 import { fileUploadMiddleware } from "./file-upload-manager";
 import { JsonStorage } from "./storage/json-storage";
 import { RedisManager } from "./redis-manager";
 import { Storage } from "./storage/storage";
 import { Response } from "express";
-const { DefaultAzureCredential } = require("@azure/identity");
-const { SecretClient } = require("@azure/keyvault-secrets");
+import rateLimit from "express-rate-limit";
+// const { DefaultAzureCredential } = require("@azure/identity");
+// const { SecretClient } = require("@azure/keyvault-secrets");
```
[the remark this answers is not in the document]
Comment (yashawasthi10) [reply api/script/default-server.ts:16]: Removed

File: api/test/redis-manager.ts (hunk 1 above)
Comment (yashawasthi10) [reply api/test/redis-manager.ts:27]: This is used in tests. Will evaluate it while open sourcing.

**Review by arpitD11 — APPROVED** [review]

**Pull request merged**